libs/openFrameworks/utils/ofRandomDistributions.h

```diff
@@ -0,0 +1,673 @@
+}
+
+// whilst nerdy enough, the above does not work with {initializer list} for glm::vec parameters such as:
+// of::random::gaussian<glm::vec2>({0,10},{1,100});
+// so this additional explicit nerdiness is required to wrap them:
+
+template<class T>
+auto gaussian(T min,  T max) -> decltype(normal<T>(min, max)) {
+  return normal(min, max);
+}
+template<class T>
+auto yes(T p) -> decltype(bernoulli(p)) {
+  return bernoulli(p);
+}
+
+// MARK: - OTHER FUNCTIONS
+
+template<class T = float> // works for all non-refined
+T bound_normal(float min, float max, float focus = 4.0f) {
+    if (min >= max) {
+        std::cout << "ofRandomNormalLimits()" << "max must be > than min\n";
+        return {};
+    } else {
+        if (focus <= .0099) {
+            std::cout << "ofRandomNormalLimits()" << "focus must be at least .01\n";
+            return {};
+        } else {
+            T v;
+            do { v = of::random::normal<T>((max+min)/2.0f, (max-min)/(2*focus)); } while (v < min || v > max);
+            return v;
+        }
+    }
+}
+
+template <typename T>
+std::enable_if_t<std::is_same_v<T, glm::vec2>, T>
+bound_normal(T min, T max, T focus = {4.0f, 4.0f}) {
+    if (!min.x < max.x || !min.y < max.y) {
+        std::cout << "ofRandomNormalLimits()" << "max must be > than min\n";
+        return {};
+    } else {
+        if (focus.x < 1 || focus.y < 1 ) {
+            std::cout << "ofRandomNormalLimits()" << "focus must be at least 1\n";
+            return {};
+        } else {
+            T v;
+            do { v.x = of::random::normal<float>((max.x+min.x)/2.0f, (max.x-min.x)/(2*focus.x)); } while (v.x < min || v.x > max);
+            do { v.y = of::random::normal<float>((max.y+min.y)/2.0f, (max.y-min.y)/(2*focus.y)); } while (v.y < min || v.y > max);
+            return v;
+        }
+    }
+}
+
+template <typename T>
+std::enable_if_t<std::is_same_v<T, glm::vec3>, T>
+bound_normal(T min, T max, T focus = {4.0f, 4.0f, 4.0f}) {
+    if (!min.x < max.x || !min.y < max.y || !min.z < max.z) {
+        std::cout << "ofRandomNormalLimits()" << "max must be > than min\n";
+        return {};
+    } else {
+        if (focus.x < 1 || focus.y < 1 || focus.z < 1) {
+            std::cout << "ofRandomNormalLimits()" << "focus must be at least 1\n";
+            return {};
+        } else {
+            T v;
+            do { v.x = of::random::normal<float>((max.x+min.x)/2.0f, (max.x-min.x)/(2*focus.x)); } while (v.x < min || v.x > max);
+            do { v.y = of::random::normal<float>((max.y+min.y)/2.0f, (max.y-min.y)/(2*focus.y)); } while (v.y < min || v.y > max);
+            do { v.z = of::random::normal<float>((max.z+min.z)/2.0f, (max.z-min.z)/(2*focus.z)); } while (v.z < min || v.z > max);
+            return v;
+        }
+    }
+}
+
+} // end namespace of::random
+
+// MARK: - ANONYMOUS NAMESPACE FUNCTIONS
+
+namespace {
+
+inline auto ofSetRandomSeed(unsigned long new_seed) {
+    of::random::seed(new_seed);
```

Comment:
should this also have the srand implementation?

Reply:
ha! actually that's a duplicate (copy/paste manipulation) as the function is provided in Math.h (where it makes more sense, alongside the other seed stuff). because i used an anonymous namespace it somehow doesn't bother the compiler... i guess it should!